weapon_bonus.py

```diff
@@ -7,245 +7,95 @@
 from get_dex_crit import *
 import numpy as np
 
-def get_weapon_bonus(main_wpn_name, rng_wpn_name, ws_name,):
-    #
-    # Check the player's main and ranged weapons for "WS Damage +x%".
-    # R15 Relics get +68% for Relic WSs
-    # R15 Mythics get +49.5% for Mythic WSs
-    # R15 Empyrean/Aeonic get +10% for Empyrean/Aeonic WSs
-    # Ambuscade weapons vary by weapon.
-    #
 
-    ws_bonus = 0
+def get_weapon_bonus(main_wpn_name, rng_wpn_name, ws_name):
+    weapon_skill_bonuses = {
+        # Main Weapons
+        "Naegling": {"Savage Blade": 0.15},
+        "Murgleis": {"Death Blossom": 0.495},
+        "Burtgang": {"Atonement": 0.3},
+        "Tizona": {"Expiacion": 0.495},
+        "Almace": {"Chant du Cygne": 0.1},
+        "Excalibur": {"Knights of Round": 0.68},
+        "Sequence": {"Requiescat": 0.1},
+        "Maxentius": {"Black Halo": 0.5},
+        "Tishtrya": {"Realmrazer": 0.1},
+        "Mjollnir": {"Randgrith": 0.68},
+        "Yagrush": {"Mystic Boon": 0.495},
+        "Idris": {"Exudation": 0.15},
+        "Epeolatry": {"Dimidiation": 0.15},
+        "Kikoku": {"Blade: Metsu": 0.68},
+        "Kannagi": {"Blade: Hi": 0.1},
+        "Nagi": {"Blade: Kamu": 0.495},
+        "Kenkonken": {"Stringing Pummel": 0.495},
+        "Heishi": {"Blade: Shun": 0.1},
```

Comment:
This should be "Heishi Shorinken" to match the "Name" key in the Heishi dictionary.

Reply:
Yes, it should. Your original code was if 'Heishi' was in the weapon name. Just going on auto-pilot, I wrote it as if it was == instead. Worth checking any other weapons you short handed as well.

Reply:
Part of my quick check was just seeing if all weapon name strings in your updated code were present as exact matches in the gear.py file's "Name" keys within the mains+ranged lists. Heishi was the only issue.

I ran a similar check against all weapon skill names in the updated code versus weapon skills selectable within the gui_wsdist.py file and found no issues, so all else should be good.